src/App.js

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import './App.css';
 
 const App = () => {
+  const initialCopy = chatMessages.map((chatEntry) => {
+    return { ...chatEntry };
+  });
+
+  const [chatData, setChatData] = useState(initialCopy);
+  const [likeCount, setLikeCount] = useState(0);
+
+  const updateLikes = (chatEntryId) => {
+    const updatedChatData = chatData.map((chatEntry) => {
+      if (chatEntry.id === chatEntryId) {
+        return {
+          ...chatEntry,
+          liked: !chatEntry.liked,
+        };
+      }
+      return chatEntry;
+    });
+    setChatData(updatedChatData);
```

Comment:
👀 Since the new state of the chat data depends on the previous state of the chat data, prefer to use the callback style of `set` function. This would rearrange the logic of calculating the new data something like the following:

```js
    setChatData(chatData => (chatData.map((chatEntry) => {
      if (chatEntry.id === chatEntryId) {
        return {
          ...chatEntry,
          liked: !chatEntry.liked,
        };
      }
      return chatEntry;
    })));
```